### PR TITLE
Filtering by event

### LINKED ISSUE #135: As a Hall admin finding residents, I want to search for students who are attending an event, so that I can quickly know the participants of an event

### PR DESCRIPTION
**Pull request by yanchenglee98**
Filtering residents by events. #135 
E.g. `filter-event 1` displays all residents attending the event 1 on the event list 
Test cases for `filter-event` will be added in the future

image

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FilterEventCommand.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        // get event from event list
+        Event event = eventList.get(eventIndex.getZeroBased());
+
+        FilterEventPredicate predicate = new FilterEventPredicate(event.getAttendeesList());
+        model.updateFilteredPersonList(predicate);
+        return new CommandResult(String.format(MESSAGE_FILTER_SUCCESS, event.getName()));
+    }
+
+    private static class FilterEventPredicate implements Predicate<Person> {
```
Comment (athuyaoo) [inline src/main/java/seedu/address/logic/commands/FilterEventCommand.java]: Why is this private static? Would it be better to separate it as another class?

**Review by athuyaoo — COMMENTED** [review]
LGTM small nit pick

File: src/main/java/seedu/address/logic/commands/FilterEventCommand.java (hunk 1 above)
Comment (yanchenglee98) [reply src/main/java/seedu/address/logic/commands/FilterEventCommand.java]: Ok will move to `seedu.address.model.event`

**Review by athuyaoo — APPROVED** [review]
lgtm

**Pull request merged**